common/src/main/java/us/myles/ViaVersion/protocols/protocol1_9to1_8/storage/EntityTracker1_9.java

```diff
@@ -97,6 +98,25 @@ public void setSecondHand(int entityID, Item item) {
         }
     }
 
+    /**
+     * It will set a shield to the offhand if a sword is in the main hand.
+     * The item in the offhand will be cleared if there is no sword in the main hand.
+     */
+    public void syncShieldWithSword() {
+        InventoryTracker inventoryTracker = getUser().get(InventoryTracker.class);
+
+        // Get item in new selected slot
+        int inventorySlot = this.heldItemSlot + 36; // Hotbar slot index to inventory slot
+        Integer itemId = inventoryTracker.getItemIdInSlot((short) inventorySlot);
+        int itemIdentifier = itemId == null ? 0 : itemId;
+
+        boolean isSword = Protocol1_9To1_8.isSword(itemIdentifier);
+        Item shield = new Item(442, (byte) 1, (short) 0, null);
+
+        // Update shield in off hand depending if a sword is in the main hand
+        setSecondHand(isSword ? shield : null);
```

Comment:
This should be short-circuited; if the itemId is null/0, set to null immediately, else also only create the shield Item instance if it actually is a sword